bleach/linkifier.py

```diff
@@ -349,7 +349,17 @@ def strip_non_url_bits(self, fragment):
 
     def handle_links(self, src_iter):
         """Handle links in character tokens"""
+        in_a = False  # happens, if parse_email=True and if a mail was found
         for token in src_iter:
+            if in_a:
+                if token['type'] == 'EndTag' and token['name'] == 'a':
+                    in_a = False
+                yield token
+                continue
+            elif token['type'] == 'StartTag' and token['name'] == 'a':
+                in_a = True
+                yield token
+                continue
```

Comment:
There's a bunch of "are we in an `<a>`" code and bookkeeping in this class. I think maybe we should adjust those cases rather than add another bookkeeping measure.

Namely, why is this method getting called if we're in an `<a>` tag at all?

Reply:
My apologies--I see what's going on now.